tests/testSpanSets.cc

```diff
@@ -0,0 +1,638 @@
+            BOOST_CHECK(output[pt.getY()-9][pt.getX()-2] == dummyArrayValue);
+        }
+    }
+    // Verify that points outside the vector have the value zero
+    BOOST_CHECK(output[0][2] == 0);
+    BOOST_CHECK(output[1][0] == 0);
+    BOOST_CHECK(output[2][2] == 0);
+
+    // Test with an array for an output
+    int outputSize = 6;
+    ndarray::Array<int, 2, 2> unflatOutput = ndarray::allocate(ndarray::makeVector(outputSize, outputSize));
+    unflatOutput.deep() = 0;
+    afwGeom::Point2I outputXY0(spnSt.getBBox().getMinX(), spnSt.getBBox().getMinY());
+    spnSt.unflatten(unflatOutput, input, outputXY0);
+
+    for (int i = 0; i < outputSize; ++i) {
+        for (int j = 0; j < outputSize; ++j) {
+            if (spnSt.contains(afwGeom::Point2I(j + outputXY0.getX(), i + outputXY0.getY()))) {
+                BOOST_CHECK(unflatOutput[i][j] == dummyArrayValue);
+            } else {
+                BOOST_CHECK(unflatOutput[i][j] == 0);
+            }
+        }
+    }
+
+    // Test with a null SpanSet
+    afwGeom::SpanSet nullSpanSet;
+    auto nullUnflatArray = nullSpanSet.unflatten(input);
+    BOOST_CHECK(nullUnflatArray.size() == 0);
+}
+
+std::pair<lsst::afw::image::Mask<lsst::afw::image::MaskPixel>, std::shared_ptr<afwGeom::SpanSet>> populateMask() {
+    // Create a mask and populate it with the value 2
+    lsst::afw::image::Mask<lsst::afw::image::MaskPixel> msk(10, 10);
+    auto spnSt = afwGeom::SpanSet::spanSetFromShape(3, afwGeom::Stencil::CIRCLE)->shiftedBy(5, 5);
+    spnSt->setMask(msk, static_cast<lsst::afw::image::MaskPixel>(2));
+    return std::pair<lsst::afw::image::Mask<lsst::afw::image::MaskPixel>,
+                     std::shared_ptr<afwGeom::SpanSet>>(msk, spnSt);
+}
+
+BOOST_AUTO_TEST_CASE(SpanSet_testSetMask) {
+    auto result = populateMask();
+    auto mask = result.first;
+    auto spnSt = result.second;
+    // Verify the mask was populated with the value 2
+    auto mskArray = mask.getArray();
+    for (auto const & spn : *spnSt) {
+        for (auto const & pt : spn) {
+            BOOST_CHECK(mskArray[pt.getY()][pt.getX()] == static_cast<lsst::afw::image::MaskPixel>(2));
+        }
+    }
+
+    // Test with a null SpanSet
+    afwGeom::SpanSet nullSpanSet;
+    lsst::afw::image::Mask<lsst::afw::image::MaskPixel> nullMask(10, 10);
+    nullSpanSet.setMask(nullMask, static_cast<lsst::afw::image::MaskPixel>(2));
+    for (auto iter = nullMask.begin(); iter != nullMask.end(); ++iter){
+        BOOST_CHECK(*iter == 0);
+    }
+}
+
+BOOST_AUTO_TEST_CASE(SpanSet_testClearMask) {
+    auto result = populateMask();
+    auto mask = result.first;
+    auto spnSt = result.second;
+    spnSt->clearMask(mask, static_cast<lsst::afw::image::MaskPixel>(2));
+    // Verify the mask is now zero
+    auto mskArray = mask.getArray();
+    for (auto const & spn : *spnSt) {
+        for (auto const & pt : spn) {
+            BOOST_CHECK(mskArray[pt.getY()][pt.getX()] == static_cast<lsst::afw::image::MaskPixel>(0));
```

Comment:
As above, there's no constraint on `clearMask` actually doing a bitwise operation. While the null case below does check that values are unchanged, it would also be helpful to test this for a nontrivial SpanSet.